crates/core/src/table/partition.rs

```diff
@@ -265,13 +265,36 @@ impl PartitionFilter {
             safe: false,
             format_options: Default::default(),
         };
+
+        let value = match data_type {
+            DataType::Date32 => Self::trim_single_quotes(value),
+            DataType::Date64 => Self::trim_single_quotes(value),
+            DataType::Utf8 => Self::trim_single_quotes(value),
+            DataType::LargeUtf8 => Self::trim_single_quotes(value),
+            DataType::Utf8View => Self::trim_single_quotes(value),
```

Comment:
time and timestamp types should be included too?

Reply:
yes it should also applies to any temporal type that is supposed to be parsed from string in sql.

Reply:
Alright, I'll add that later today